runtime/tests/org.eclipse.core.tests.harness/src/org/eclipse/core/tests/harness/FileSystemComparator.java

```diff
@@ -123,17 +123,15 @@ public Object loadSnapshot(File rootLocation) throws IOException {
 
 	public void saveSnapshot(Object toSave, File rootLocation) throws IOException {
 		File summaryFile = new File(rootLocation, SNAPSHOT_FILE_NAME);
-		PrintWriter out = new PrintWriter(new OutputStreamWriter(new BufferedOutputStream(new FileOutputStream(summaryFile))));
-		Map<?, ?> snapshot = (Map<?, ?>) toSave;
-		try {
+		try (PrintWriter out = new PrintWriter(
+				new OutputStreamWriter(new BufferedOutputStream(new FileOutputStream(summaryFile))))) {
```

Comment:
What "StreamEncoder" are you talking about?
All close() are supposed to fail only on a hard jdk Error - like out of memory. In such a case a clean continuation can not be guaranteed anyway. 
see java.lang.AutoCloseable.close() javadoc:
```
Cases where the close operation may fail require careful
     * attention by implementers. It is strongly advised to relinquish
     * the underlying resources and to internally <em>mark</em> the
     * resource as closed, prior to throwing the exception. The {@code
     * close} method is unlikely to be invoked more than once and so
     * this ensures that the resources are released in a timely manner.
     * Furthermore it reduces problems that could arise when the resource
     * wraps, or is wrapped, by another resource.
```

Reply:
`OutputStreamWriter` uses a `StreamEncoder` for encoding output into a byte stream. The `OutputStreamWriter` is _not_ an `AutoClosable` and will eventually fail during its `Closable#close()` without its output stream being closed, because the `StreamEncoder` flushes a buffer by trying to write to the file which may fail and leave the underlying `FileOutputStream` open.
Since this is part of the test harness, it could result in a "dangling" file handle that may conflict with further test executions.

Reply:
please doublecheck: the OutputStreamWriter is AutoClosable 
The internally used StreamEncoder will only fail with IOException if the underlying stream throws an IOException during write - in that case a close() is not need anymore.  If you find a counter-proof please report it to JDK.

Reply:
You are right, sorry for the confusion.